### DISCUSSION TITLE
REST API v3: Not able to create repo using the code given in the guide

### DISCUSSION THREAD

**Discussion by Nirajn2311** [community/community#24385] [Apps, API and Webhooks]
<p>I used the code given here - <a href="https://developer.github.com/v3/guides/getting-started/#create-a-repository" rel="nofollow noopener">https://developer.github.com/v3/guides/getting-started/#create-a-repository</a> - to create a repo using REST API v3. But it errors out with a message <code>Problems parsing JSON</code>. I have even replaced the OAuth token with my own. Can someone help me out?</p>

**Comment by abinoda** [comment]
<p>Can you post your exact code or curl example (with OAuth tokens redacted) so we can see what you’re doing? Difficult to provide feedback without that.</p>

**Comment by Nirajn2311** [comment]
<p><a class="mention" href="/abinoda">@abinoda</a> Heres the code -</p>
<pre><code>curl -i -H "Authorization: token \&lt;OAuth token\&gt;" \ -d '{ \
        "name": "blog", \
        "auto_init": true, \
        "private": true, \
        "gitignore_template": "nanoc" \
      }' \
    https://api.github.com/user/repos
</code></pre>
<p>The above code gets the parsing JSON error.</p>
<p>But if I put everything in one line then the request works</p>
<p>The one line code - </p>
<pre><code>curl -i -H "Authorization: token &lt;OAuth token&gt;" -d '{"name": "blog", "auto_init": true, "private": true, "gitignore_template": "nanoc"}' https://api.github.com/user/repos`
</code></pre>

**Comment by abinoda** [comment, marked as answer]
<p>Try this, I think it’s cleaner:</p>
<pre><code>curl -i -X POST https://api.github.com/user/repos \
-H "Authorization: token &lt;OAuth token&gt;" \
-d @- &lt;&lt; EOF
{
  "name": "blog",
  "auto_init": true,
  "private": true,
  "gitignore_template": "nanoc"
}
EOF
</code></pre>

**Comment by Nirajn2311** [comment]
<p> <a class="mention" href="/abinoda">@abinoda</a> That worked, thx for the help.</p>
<p>I think that the guide should be updated as this was not the only code that had the same error.</p>

**Comment by ghost** [comment]
<p>Can i pass parameters like “created_at” “updated_at” “pushed_at” ? It says problems parsing json,</p>

**Comment by soydepr** [comment]
<p>For me it only works as single line after many tries</p>
